search-impl/src/main/java/com/example/auction/search/Module.java

```diff
@@ -0,0 +1,30 @@
+package com.example.auction.search;
+
+import com.example.auction.bidding.api.BiddingService;
+import com.example.auction.item.api.ItemService;
+import com.example.auction.search.api.SearchService;
+import com.example.auction.search.impl.BrokerEventConsumer;
+import com.example.auction.search.impl.IndexedStoreImpl;
+import com.example.auction.search.impl.SearchServiceImpl;
+import com.example.elasticsearch.Elasticsearch;
+import com.google.inject.AbstractModule;
+import com.lightbend.lagom.javadsl.server.ServiceGuiceSupport;
+
+/**
+ *
+ */
+public class Module extends AbstractModule implements ServiceGuiceSupport {
+    @Override
+    protected void configure() {
+        bindServices(serviceBinding(SearchService.class, SearchServiceImpl.class));
+
+        bindClient(BiddingService.class);
+        bindClient(ItemService.class);
+        bindClient(Elasticsearch.class);
+
+        bind(IndexedStore.class).to(IndexedStoreImpl.class);
+
+        bind(BrokerEventConsumer.class).asEagerSingleton();
+
```

Comment:
I'm not sure I'm following your reasoning. I wanted to make it obvious in some way or another that `searchserrvice-impl` consumes from two topics.
In this case in particular events incoming from the topics will go straight into the `IndexedStoreImpl` but in other services the event will act as commands for a Persistent Entity.

At the moment, the responsibilities of `BrokerEventConsumer` are building the stream and adapting the public API's of upstream services into `IndexedStore`'s API. There's no intermediate representation of the data but at least  `IndexedStore` doesn't know anything about upstream API's.

Maybe what we need is for `IndexedStoreImpl` to provide a Sink we can plug the `Source`'s into.